Source/scrollrt.cpp

```diff
@@ -318,9 +319,10 @@ void DrawMissilePrivate(const Surface &out, const Missile &missile, Point target
 
 	const Point missileRenderPosition { targetBufferPosition + missile.position.offsetForRendering - Displacement { missile._miAnimWidth2, 0 } };
 	CelSprite cel { missile._miAnimData, missile._miAnimWidth };
-	if (missile._miUniqTrans != 0)
-		Cl2DrawLightTbl(out, missileRenderPosition.x, missileRenderPosition.y, cel, missile._miAnimFrame, missile._miUniqTrans + 3);
-	else if (missile._miLightFlag)
+	if (missile._miUniqTrans != 0) {
+		uint8_t *trn = GetUniqueMonsterTRN(UniqueMonstersData[Monsters[missile._misource]._uniqtype - 1].mTrnName);

```

Comment:
This does a string map lookup in the render loop. Can't we load these once at initialization time and get rid of the map altogether?

Reply:
will try :P

Reply:
Added TRNs to monsters (at least I hope so) @glebm